### PR TITLE
[ENH] Implemented mapie>1.0 interface

### LINKED ISSUE #563: [ENH] `mapie 1.X` compatibility
The `MapieRegressor` is no longer compatible with `mapie` since version 1. The upstream regressor has been split into four classes.

We should:

* add the `mapie<1.0` dependency to the `MapieRegressor`
* add four classes interfacing the four `mapie` regression classes

### PR DESCRIPTION
**Pull request by Omswastik-11**
## Overview

This PR adds support for **MAPIE ≥ 1.0** by introducing new wrapper classes for the updated MAPIE regression API.
MAPIE 1.0 split the original `MapieRegressor` into several specialized estimators, so skpro now provides dedicated interfaces for each.

---

### New Files

* **`skpro/regression/conformal/_mapie_split_conformal.py`** — Split Conformal Regressor
* **`skpro/regression/conformal/_mapie_cross_conformal.py`** — Cross Conformal Regressor
* **`skpro/regression/conformal/_mapie_cqr.py`** — Conformalized Quantile Regressor
* **`skpro/regression/jackknife/_mapie_jackknife.py`** — Jackknife After Bootstrap Regressor
* **`skpro/regression/conformal/__init__.py`** — Module init for conformal regressors
* **`skpro/regression/jackknife/__init__.py`** — Module init for jackknife regressors
* **`skpro/regression/tests/test_mapie_v1.py`** — Test suite for the new wrappers

### Modified Files

* **`skpro/regression/__init__.py`** — Exports new MAPIE classes
* **`docs/source/api_reference/regression.rst`** — API documentation for new modules

---

### Added four new probabilistic regressors:

1. **`MapieSplitConformalRegressor`**
   Split conformal prediction using a train/calibration split

2. **`MapieCrossConformalRegressor`**
   Cross-validation–based conformal prediction (Jackknife+ variants)

3. **`MapieJackknifeAfterBootstrapRegressor`**
   Jackknife+ after bootstrap

4. **`MapieConformalizedQuantileRegressor`**
   Conformalized Quantile Regression (CQR)

### All new classes:

* Pass `check_estimator` validation
* Handle mixed column name types correctly via `prep_skl_df`
* Convert `y` to 1D array to avoid `DataConversionWarning`
* Tagged with:

  ```python
  python_dependencies = ["MAPIE>=1.0"]


### API Usage Examples
----
1. Split Conformal Regressor

```python
from skpro.regression.conformal import MapieSplitConformalRegressor
from sklearn.linear_model import LinearRegression

reg = MapieSplitConformalRegressor(
    estimator=LinearRegression(),
    test_size=0.3,
    method="absolute",  # or "gamma"
    random_state=42,
)
reg.fit(X_train, y_train)

y_pred = reg.predict(X_test)
y_pred_interval = reg.predict_interval(X_test, coverage=[0.9, 0.95])

```

2.  Cross Conformal Regressor

```python

from skpro.regression.conformal import MapieCrossConformalRegressor
from sklearn.linear_model import LinearRegression

reg = MapieCrossConformalRegressor(
    estimator=LinearRegression(),
    cv=5,
    method="plus",  # or "minmax"
    random_state=42,
)
reg.fit(X_train, y_train)

y_pred_interval = reg.predict_interval(X_test, coverage=0.9)

```

3.  Jackknife After Bootstrap

```python

from skpro.regression.jackknife import MapieJackknifeAfterBootstrapRegressor
from sklearn.linear_model import LinearRegression

reg = MapieJackknifeAfterBootstrapRegressor(
    estimator=LinearRegression(),
    cv=50,               # number of bootstrap resamples
    method="plus",
    agg_function="mean", # or "median"
    random_state=42,
)
reg.fit(X_train, y_train)

y_pred_interval = reg.predict_interval(X_test, coverage=0.9)

```

4.  Conformalized Quantile Regression
```python
from skpro.regression.conformal import MapieConformalizedQuantileRegressor
from sklearn.ensemble import GradientBoostingRegressor

reg = MapieConformalizedQuantileRegressor(
    estimator=GradientBoostingRegressor(loss="quantile", alpha=0.5),
    confidence_level=0.9,
    test_size=0.3,
    random_state=42,
)
reg.fit(X_train, y_train)

y_pred_interval = reg.predict_interval(X_test, coverage=0.9)

```

Fixes #563

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Very nice!

Small changes only:

* please add the mapie authors as authors to the `authors` field (only those with significant contributions to the regressors)
* can you add the new regressors to the API reference docs too, please?

I would also suggest to not put the estimators in a single file, but move these as follows:

* new module `conformal` for all estimators except the jacknnife-after bootstrap. This contains private files like `_mapie_split_conforma.py` which have one estimator each, and are exported in `__init__`
* new module `jackknife` (same folder + private module) which gets the new jackknife regressor, and later (not this PR) we could also move enbpi there.
